test/reg_sort_test.rb

```diff
@@ -0,0 +1,45 @@
+# encoding: ASCII-8BIT
+require 'test_helper'
+require 'pwnlib/reg_sort'
+
+class RegSortTest < MiniTest::Test
+  include ::Pwnlib::RegSort::ClassMethod
+
+  def test_check_cycle
+    assert_equal([], check_cycle('a', a: 1))
+    assert_equal(['a'], check_cycle('a', a: 'a'))
+    assert_equal([], check_cycle('a', a: 'b', b: 'c', c: 'b', d: 'a'))
+    assert_equal(%w(a b c d), check_cycle('a', a: 'b', b: 'c', c: 'd', d: 'a'))
+  end
+
+  def test_regsort
```

Comment:
Should we break this function into many that test different arguments?
test_regsort_tmp, test_regsort_post_move, ...